### ISSUE TITLE
Restart required to attach to new input devices

### ISSUE THREAD

**Issue by christf** [josh-richardson/cadmus#47]
The title says it all, I guess.

When I 

1. start cadmus
2. attach a USB microphone

I am unable to denoise that microphone via cadmus as it is not displayed as a possible source. After killing and re-starting cadmus this works.

On that note, the list of inputs seems to have other issues as well: a certain "Denoised microphone" turns up here which is unexpected as I do not thnk cadmus should daisy-chain the denoise filterson top of each other.

**Comment by josh-richardson** [comment]
Yes, Cadmus doesn't currently listen to events to see if new devices are added and removed. This is a possible enhancement.
